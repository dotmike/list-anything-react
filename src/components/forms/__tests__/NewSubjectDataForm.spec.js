import configureStore from "redux-mock-store";
import React from "react";

import ConnectedNewSubjectDataForm, {
  UnconnectedNewSubjectDataForm
} from "../NewSubjectDataForm";

const defaultProps = {
  submit: () => {}
};

describe("ConnectedNewSubjectDataForm", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(
      <ConnectedNewSubjectDataForm {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedNewSubjectDataForm", () => {
  const props = {
    ...defaultProps,
    fields: [],
    tabs: []
  };

  it("renders correctly", () => {
    props.hasSubjects = false;

    const wrapper = shallowWithContext(
      <UnconnectedNewSubjectDataForm {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
