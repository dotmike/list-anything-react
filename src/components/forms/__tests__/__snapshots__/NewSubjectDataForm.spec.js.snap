// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedNewSubjectDataForm renders correctly 1`] = `
<NewSubjectDataForm
  dispatch={[Function]}
  fields={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  submit={[Function]}
  tabs={Array []}
/>
`;

exports[`UnconnectedNewSubjectDataForm renders correctly 1`] = `
<Form
  as="form"
  error={false}
  loading={false}
  onSubmit={[Function]}
  size="large"
>
  <Segment
    stacked={true}
    textAlign="left"
  >
    <FormField
      control="select"
      label="Select the tab:"
      name="tabId"
      onChange={[Function]}
      value=""
    />
    <Button
      as="button"
      color="teal"
      fluid={true}
      size="large"
    >
      Add
    </Button>
  </Segment>
</Form>
`;
